src/utils/Recorder.ts

```diff
@@ -0,0 +1,128 @@
+import { MockedRequest } from './handlers/requestHandler'
+import { fetch } from '../context/fetch'
+import { SetupApi, RequestHandlersList } from '../setupWorker/glossary'
+import { rest, restContext } from '../rest'
+import { ResponseComposition } from '../response'
+
+type RecordRequest = {
+  function: string
+  returnStatement: string
+  method: Request['method']
+  url: URL
+}
+
+async function createRecordFromRequest(
+  request: MockedRequest,
+  response: Response,
+): Promise<RecordRequest> {
+  const lines = []
+  const composeLines = []
+  lines.push(
+    `rest.${request.method.toLowerCase()}('${
+      request.url
+    }',function(req,res,ctx){`,
+  )
+
+  composeLines.push(`ctx.status(${response.status})`)
+  const body = await response.text()
+
+  if (body) {
+    const hasJsonContent = response.headers
+      ?.get('content-type')
+      ?.includes('json')
+    if (hasJsonContent) {
+      composeLines.push(`ctx.json(${body})`)
+    } else {
+      composeLines.push(`ctx.body(\`${body.replace(/`/g, '\\`')}\`)`)
+    }
+  }
```

Comment:
How we can handle binary entities ? To record the API , we should include the binary in the created function, but it will be not very clean for the user